docs/iris/src/whatsnew/collate_contributions.py

```diff
@@ -0,0 +1,160 @@
+# (C) British Crown Copyright 2015, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris. If not, see <http://www.gnu.org/licenses/>.
+
+import datetime
+import os
+import sys
+import re
+from operator import itemgetter
+from distutils import version
+from __future__ import (absolute_import, division, print_function)
+
+# Regular expressions: CONTRIBUTION_REGEX matches the filenames of
+# contribution snippets. It is split into three sections separated by _
+# 0. String for the category. 1. ISO8601 date. 2. String for the feature name.
+# RELEASE_REGEX matches the directory names, returning the release.
+CONTRIBUTION_REGEX = re.compile(r'(.*)\_(\d{4}-\d{2}-\d{2})\_(.*)\.txt$')
+RELEASE_REGEX = re.compile(r'.*\_(.*)$')
+SOFTWARE_NAME = 'Iris'
+VALID_CATEGORIES = [
+    {'Prefix': 'newfeature', 'Title': 'Features'},
+    {'Prefix': 'bugfix', 'Title': 'Bugs Fixed'},
+    {'Prefix': 'incompatiblechange', 'Title': 'Incompatible Changes'},
+    {'Prefix': 'deprecate', 'Title': 'Deprecations'},
+    {'Prefix': 'docchange', 'Title': 'Documentation Changes'}
+]
+
+
+def get_release_number(directory_name):
+    '''Takes a formatted string and returns a LooseVersion object.'''
+    compilable_files_unsorted = []
+    directory_elements = RELEASE_REGEX.findall(directory_name)
+    try:
+        return version.LooseVersion(directory_elements[0])
+    except IndexError:
+        return None
+
+
+def get_directory_list(root_directory):
+    '''Returns a list of subdirectories, excluding files.'''
+    directory_contents = os.listdir(root_directory)
+    subdirectories = []
+    for inode in directory_contents:
+        if os.path.isdir(inode):
+            subdirectories.append(inode)
+    return subdirectories
+
+
+def check_compiled(directory_name):
+    '''Checks if a release.rst file exists for each contribution directory.'''
+    extension = '.rst'
+    release = get_release_number(directory_name)
+    compiled_filename = str(release) + extension
+    print compiled_filename
+    if os.path.exists(compiled_filename):
+        if os.path.isfile(compiled_filename):
+            return True
+    return False
```

Comment:
Multiple returns! Returning a result that is either `True` or `False` is better.